sdk/python/feast/entity.py

```diff
@@ -44,3 +55,273 @@ def from_proto(cls, entity_proto: EntityProto):
         """
         entity = cls(name=entity_proto.name, dtype=ValueType(entity_proto.value_type))
         return entity
+
+
+class EntityV2:
```

Comment:
Do we really need to expose the `V2` to users? I think we can replace the existing Entity with this new one. Any concerns with that @mrzzy ?

Reply:
I think that's okay, but we'll need to update (perhaps use another name?) the previous Entity implementation used in tests and feature set. However, this would introduce breaking change to existing users who are still using the featureset-entity concept.

eg. if they do something like
```
fs = FeatureSet(
        "featureset",
        features=[
            Feature("feat1", ValueType.STRING)
        ],
        entities=[Entity("entity_id", ValueType.INT64)],
        max_age=Duration(seconds=100),
    )
```

Reply:
Will be updating SDK in separate PR, along with replacing existing e2e tests.